http-server/src/server.rs

```diff
@@ -170,11 +170,20 @@ impl<M> Builder<M> {
 	}
 
 	/// Enable health endpoint.
-	/// Allows you to expose one of the methods under GET /<path> The method will be invoked with no parameters. Error returned from the method will be converted to status 500 response.
-	/// Expects a tuple with (<path>, <rpc-method-name>).
-	pub fn health_api(mut self, path: impl Into<String>, method: impl Into<String>) -> Self {
-		self.health_api = Some(HealthApi { path: path.into(), method: method.into() });
-		self
+	/// Allows you to expose one of the methods under GET /<path> The method will be invoked with no parameters.
+	/// Error returned from the method will be converted to status 500 response.
+	/// Expects a tuple with (</path>, <rpc-method-name>).
+	///
+	/// Fails if the path is missing `/`.
+	pub fn health_api(mut self, path: impl Into<String>, method: impl Into<String>) -> Result<Self, Error> {
+		let path = path.into();
+
+		if !path.starts_with("/") {
+			return Err(Error::Custom(format!("Health endpoint path must start with `/` to work, got: {}", path)));
+		}
```

Comment:
Just wondering; why do we care about the path starting with '/'?

Reply:
Because we have [this check](https://github.com/paritytech/jsonrpsee/blob/master/http-server/src/server.rs#L458), if `/` is missing that will fail and it's quite tricky to debug.

I have already done that myself but more ideally would be `Path type` instead of a String

Reply:
Ah ok gotcha; that makes sense!